src/main/java/com/limechain/network/protocol/blockannounce/BlockAnnounceEngine.java

```diff
@@ -68,6 +68,7 @@ private void handleBlockAnnounce(byte[] msg, PeerId peerId) {
                 " with hash:0x" + HexUtils.toHexString(announce.getHeader().getHash()) +
                 " parentHash:" + announce.getHeader().getParentHash() +
                 " stateRoot:" + announce.getHeader().getStateRoot());
+        //TODO: Should update Trie (and merkle proofs?) and save them in db
```

Comment:
_**I think that**_ if the state root of the header has changed the host should send out a state request for the block hash(since light client can't process extrinsics) and update it's trie.

IMO this should be in a separate task in order to not bloat this PR

Reply:
I agree that it should be in separate task, that's why I left it with todo